src/Data/HashTable/ST/Basic.hs

```diff
@@ -620,6 +681,12 @@ recordIsDeleted = (== deletedMarker)
 
 
 ------------------------------------------------------------------------------
+{-# INLINE recordIsFilled #-}
+recordIsFilled :: Elem -> Bool
+recordIsFilled el = not (recordIsEmpty el) && not (recordIsDeleted el)
```

Comment:
Now I see. I should have noticed that the type signature was different.

Then how about this?
```haskell
recordIsFilled :: Elem -> Bool
recordIsFilled !el = tagToEnum# isFilled#                                           
  where
    !el# = U.elemToInt# el
    !deletedMarker# = U.elemToInt# deletedMarker                                    
    !emptyMarker# = U.elemToInt# emptyMarker                                        
#if __GLASGOW_HASKELL__ >= 708
    !isFilled# = (el# /=# deletedMarker#) `andI#` (el# /=# emptyMarker#)            
#else
    !delOrEmpty# = mask# el# deletedMarker# `orI#` mask# el# emptyMarker#           
    !isFilled# = 1# `andI#` notI# delOrEmpty#                                       
#endif
```

Sorry for the overabundance of questions, I'm just trying to minimize the number of commits I make before I get this patch right. :)